components/_patterns/02-molecules/menus/tabs/tabs.js

```diff
@@ -1,54 +1,61 @@
-(function () {
+/**
+ * @file
+ * A JavaScript file containing the main menu functionality (small/large screen)
+ *
+ */
 
-  'use strict';
+// JavaScript should be made compatible with libraries other than jQuery by
+// wrapping it with an "anonymous closure". See:
+// - https://drupal.org/node/1446420
+// - http://www.adequatelygood.com/2010/3/JavaScript-Module-Pattern-In-Depth
 
-  var el = document.querySelectorAll('.tabs');
-  var tabNavigationLinks = document.querySelectorAll('.tabs__link');
-  var tabContentContainers = document.querySelectorAll('.tabs__tab');
-  var activeIndex = 0;
-
-  /**
-   * handleClick
-   * @description Handles click event listeners on each of the links in the
-   *   tab navigation. Returns nothing.
-   * @param {HTMLElement} link The link to listen for events on
-   * @param {Number} index The index of that link
-   */
-  var handleClick = function (link, index) {
-    link.addEventListener('click', function (e) {
-      e.preventDefault();
-      goToTab(index);
-    });
-  };
+(function tabs() {
+  const el = document.querySelectorAll('.tabs');
+  const tabNavigationLinks = document.querySelectorAll('.tabs__link');
+  const tabContentContainers = document.querySelectorAll('.tabs__tab');
+  let activeIndex = 0;
 
   /**
    * goToTab
    * @description Goes to a specific tab based on index. Returns nothing.
    * @param {Number} index The index of the tab to go to
    */
-  var goToTab = function (index) {
+  function goToTab(index) { // eslint-disable-line func-names
```

Comment:
I've seen this eslint rule disabled several times in the diff here. You might consider setting that config value globally, depending on what you're standardizing on. [Here](http://2ality.com/2015/09/function-names-es6.html) is a helpful article on function names in ES6. I don't personally see a problem with defining functions in this way.

Reply:
I think these must have been cruft cause they're named. I'd rather keep that setting and only ignore it in special cases.